### PR TITLE
Multi-gpu improvements

### PR DESCRIPTION
**Pull request by ethanweber**
[DDP](https://pytorch.org/tutorials/beginner/dist_overview.html) seems to be working, but it only really makes sense for long or compute-heavy training runs. Here is a small test for only 1K iterations with vanilla-nerf.

```python
# 3 GPU (3x LR)
ns-train vanilla-nerf --machine.num-gpus 3 --vis wandb --trainer.max-num-iterations 1000 --optimizers.fields.optimizer.lr 15e-4 --trainer.steps-per-eval-batch 100 --trainer.steps-per-eval-image 100 --logging.local-writer.max-log-size=0

# 1 GPU (1x LR)
ns-train vanilla-nerf --machine.num-gpus 1 --vis wandb --trainer.max-num-iterations 1000 --optimizers.fields.optimizer.lr 5e-4 --trainer.steps-per-eval-batch 100 --trainer.steps-per-eval-image 100 --logging.local-writer.max-log-size=0
```

image

### CODE REVIEW

**Review by tancik — APPROVED** [review]
LGTM

**Comment by ethanweber** [comment]
Note that further experiments are needed to verify that the code works as expected. The key would be to find a high learning rate where the model diverges on 1 GPU but works with multiple GPUs.
